source/adios2/core/IO.tcc

```diff
@@ -260,6 +238,66 @@ ADIOS2_FOREACH_STDTYPE_2ARGS(make_GetVariableMap)
 ADIOS2_FOREACH_ATTRIBUTE_STDTYPE_2ARGS(make_GetAttributeMap)
 #undef make_GetAttributeMap
 
+template <class T>
+Params IO::GetVariableInfo(const std::string &variableName,
+                           const std::set<std::string> &keys)
+{
+    Params info;
+    // keys input are case insensitive
+    const std::set<std::string> keysLC = helper::LowerCase(keys);
+
+    // return empty map if only "name" key is requested
+    if (keys.size() == 1 && keysLC.count("name") == 1)
+    {
+        return info;
+    }
+
+    Variable<T> &variable = *InquireVariable<T>(variableName);
+
+    if (keys.empty() || keysLC.count("type") == 1)
+    {
+        info["Type"] = variable.m_Type;
+    }
+
+    if (keys.empty() || keysLC.count("availablestepscount") == 1)
+    {
+        info["AvailableStepsCount"] =
+            helper::ValueToString(variable.m_AvailableStepsCount);
+    }
+
+    if (keys.empty() || keysLC.count("shape") == 1)
+    {
+        // expensive function
+        info["Shape"] = helper::VectorToCSV(variable.Shape());
+    }
+
+    if (variable.m_SingleValue)
+    {
+        if (keys.empty() || keysLC.count("singlevalue") == 1)
```

Comment:
Quick question: wasn't min/max set to the value of the single value previously? With this implementation now you cannot get the actual value with available_variables(), right?

Reply:
Correct, that's what the `Variable::Min()` and `Variable::Max()` would return, but it was never exposed in the `AvailableVariables` front ends. This implementation shouldn't be changing that behavior to keep APIs backwards compatible, unless I made a mistake.

Reply:
I think, now even if you ask for min/max or for everything with an empty set, single values will have no entry in the map for min and max. So maybe you need to add min/max in the true branch of that condition.

Reply:
OK, I see. I guess it won't change current behavior. Let me add that.

Reply:
@pnorbert should we return Max/Min for single values if the keys are empty? (warning: that would modify the returned dict in `v2.5.0`).

Reply:
Oh, I now realize we did not give back the actual value in any form in this dictionary before. I think it would be useful to extend then the dictionary with a min/max value for them as well. 

Here is the pixie3d.bp data, an array,  a value over time, and a single string value for the entire 
runtime, with bpls listing:

```
  double   /Car_variables/Bz               11*{2, 33, 33} = 0.999978 / 1.00002
  int32_t  itime                           11*scalar = 0 / 10
  string   visit_expressions               scalar = "V:{<Car_variables/Vx>,<Car_variables/Vy>,<Car_variables/Vz>} ; B:{<Car_variables/Bx>,<Car_variables/By>,<Car_variables/Bz>} ; J:{<Car_variables/Jx>,<Car_variables/Jy>,<Car_variables/Jz>} ; Ve:{<Car_variables/Ve_1>,<Car_variables/Ve_2>,<Car_variables/Ve_3>}"
``` 

In python, right now the dictionary has no info on their value(s) except for the array:

```
>>> v['/Car_variables/Bz']
{'AvailableStepsCount': '11', 'Max': '1.00002', 'Min': '0.999978', 'Shape': '2, 33, 33', 'SingleValue': 'false', 'Type': 'double'}
>>> v['itime']
{'AvailableStepsCount': '11', 'Shape': '', 'SingleValue': 'true', 'Type': 'int32_t'}
>>> v['visit_expressions']
{'AvailableStepsCount': '1', 'Shape': '', 'SingleValue': 'true', 'Type': 'string'}
>>> 
```
 
I think it would be nice to see them for the values as well.

Reply:
OK, will do. I remember now, this dict predates the Min Max functions that get updated on each step. There was a bug for Value which I then removed as it's not the canonical way (Get is) to get actual global values.